### PR TITLE
[PAY-890] Add locked content modal

### PR DESCRIPTION
**Pull request by sddioulde**
### Description

Adds modal that displays when a user clicks on a premium track tile they don't have access to.

https://github.com/AudiusProject/audius-client/pull/2910 will merge into this one.

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

local dapp against stage. tested display of modal and unlocking of content within modal.

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

existing

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sd-pc-locked-modal

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sd-pc-locked-modal

File: packages/web/src/components/track/GiantTrackTile.module.css [hunk 1]
```diff
@@ -205,6 +205,7 @@
   display: flex;
   align-items: flex-start;
   justify-content: space-between;
+  gap: 16px;
```
Comment (dharit-tan) [inline packages/web/src/components/track/GiantTrackTile.module.css:208]: Nice

File: packages/web/src/components/track/PremiumTrackSection.tsx [hunk 2]
```diff
@@ -84,13 +84,17 @@ type PremiumTrackAccessSectionProps = {
...
+  buttonClassName?: string
 }
 
 const LockedPremiumTrackSection = ({
   premiumConditions,
   followee,
   tippedUser,
-  goToCollection
+  goToCollection,
+  className,
```
Comment (dharit-tan) [inline packages/web/src/components/track/PremiumTrackSection.tsx:96]: Nit: should be named `style` or something instead?

Comment (dharit-tan) [reply packages/web/src/components/track/PremiumTrackSection.tsx:96]: And buttonStyle etc.

**Review by dharit-tan — APPROVED** [review]

File: packages/web/src/components/track/PremiumTrackSection.tsx (hunk 2 above)
Comment (sddioulde) [reply packages/web/src/components/track/PremiumTrackSection.tsx:96]: on web we've mostly used `className` actually. there are 3 instance of `style` with web but we should do our best to be consistent. maybe we change those to `className` too

File: packages/web/src/components/track/GiantTrackTile.module.css (hunk 1 above)
Comment (sddioulde) [reply packages/web/src/components/track/GiantTrackTile.module.css:208]: yup!

**Pull request merged**